### PR TITLE
feat: typings support all imports

### PR DESCRIPTION
**Pull request by christophehurpeau**

### CODE REVIEW

**Comment by mattbishop** [comment]
This looks better than mine. 👍

**Comment by christophehurpeau** [comment]
@mattbishop sorry I forgot the typings folder 🤦‍♂️

File: typings/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "target": "esnext",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
```
Comment (mattbishop) [inline typings/tsconfig.json]: JSON isn't allowed to have comments.

File: typings/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    "module": "esnext",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
+    "lib": ["dom", "es5", "es6", "es2015", "es7", "es2016", "es2017", "es2018", "esnext"], /* Polyfills are imported either by babel or with polyfill.io */
```
Comment (mattbishop) [inline typings/tsconfig.json]: None of these libs are required. this line can be removed.

File: typings/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "compilerOptions": {
+    "target": "esnext",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "module": "esnext",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
```
Comment (mattbishop) [inline typings/tsconfig.json]: This should be "none" as a module is not being generated.

File: typings/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "lib": ["dom", "es5", "es6", "es2015", "es7", "es2016", "es2017", "es2018", "esnext"], /* Polyfills are imported either by babel or with polyfill.io */
+    "isolatedModules": true,                  /* Transpile each file as a separate module (similar to 'ts.transpileModule'). */
+    "noEmit": true,
+
+    /* Module Resolution Options */
+    "moduleResolution": "node",               /* Specify module resolution strategy: 'node' (Node.js) or 'classic' (TypeScript pre-1.6). */
```
Comment (mattbishop) [inline typings/tsconfig.json]: moduleResolution can be removed.

Comment (christophehurpeau) [reply typings/tsconfig.json]: without this, I have an error:
```
typings/import-default-should-pass.ts:1:24 - error TS2307: Cannot find module '..'.

1 import deepFreeze from "..";
```

File: typings/tsconfig.json (hunk 1 above)
Comment (christophehurpeau) [reply typings/tsconfig.json]: comments are allowed in tsconfig.json: https://github.com/Microsoft/TypeScript/pull/5450

Comment (mattbishop) [reply typings/tsconfig.json]: Somebody needs to tell github ;)

**Pull request merged**